src/main/java/hudson/tasks/junit/pipeline/JUnitResultsStepExecution.java

```diff
@@ -50,8 +52,9 @@ protected TestResultSummary run() throws Exception {
             TestResultAction testResultAction = JUnitResultArchiver.parseAndAttach(step, pipelineTestDetails, run, workspace, launcher, listener);
 
             if (testResultAction != null) {
-                // TODO: Once JENKINS-43995 lands, update this to set the step status instead of the entire build.
-                if (testResultAction.getResult().getFailCount() > 0) {
+                int testFailures = testResultAction.getResult().getResultByNode(nodeId).getFailCount();
```

Comment:
I guess the insertion of the `getResultByNode` call here is intentional.

It means you can introduce a variable and use it on line 60.

Reply:
Good catch, done in 030f0db.